packages/api/src/services/lsp-state.ts

```diff
@@ -0,0 +1,129 @@
+import * as uma from "@uma/sdk";
+import { BatchRead, toString, toNumber, parseBytes, nowS } from "../libs/utils";
+import { AppState } from "..";
+import { lsps } from "../tables";
+
+type Instance = uma.clients.lsp.Instance;
+type Config = undefined;
+type Dependencies = Pick<
+  AppState,
+  "lsps" | "registeredLsps" | "provider" | "collateralAddresses" | "shortAddresses" | "longAddresses" | "multicall"
+>;
+
+export default (config: Config, appState: Dependencies) => {
+  const { lsps, registeredLsps, provider, collateralAddresses, shortAddresses, longAddresses, multicall } = appState;
+
+  // default props we want to query on contract
+  const staticProps: [string, (x: any) => any][] = [
+    // TODO: deal with this when early contracts dont have pairName
+    // ["pairName", (x:any)=>an],
+    ["collateralPerPair", toString],
+    ["priceIdentifier", parseBytes],
+    ["collateralToken", toString],
+    ["longToken", toString],
+    ["shortToken", toString],
+    ["finder", toString],
+    ["financialProductLibrary", toString],
+    ["customAncillaryData", toString],
+    ["prepaidProposerReward", toString],
+  ];
+
+  const dynamicProps: [string, (x: any) => any][] = [
+    ["expiryPrice", toString],
```

Comment:
how is `expiryPrice` and `expirationTimestamp` dynamic? once it is set it wont change, unlike contract state or `expiryPercentLong` which can change during the life of an LSP.

Reply:
also, there is a state variable `expiryPrice` which might be useful to store

Reply:
im considering static props set on contract creation, and dont change after that. I will move those to be static, good call. Just want to double check on `expiryPrice`, it doesnt seem like something that would be set at start of contract. 

> also, there is a state variable expiryPrice which might be useful to store

did you mean `expiryPercentLong`? it doesnt look like i have that one

Reply:
oh nm i see it